Initial criteria for models, embeddings (#4223)
spiceai/spiceai: docs/criteria/embeddings/stable.md

# Spice.ai OSS Embeddings - Stable Release Criteria

This document defines the set of criteria that is required before an embedding component is considered to be of Stable quality.

All criteria must be met for the embedding component to be considered Stable.

## Stable Quality Embeddings

| Embedding Type | Stable Quality | DRI Sign-off |
| -------------- | -------------- | ------------ |
| File           | ➖             |              |
| Hugging Face   | ➖             |              |

> | Spice.ai       | ➖             |              |
> | OpenAI         | ➖             |              |
> | Azure          | ➖             |              |
> | Grok           | ➖             |              |

Reviewer: ```suggestion
| Spice.ai Cloud Platform       | ➖             |              |
| OpenAI         | ➖             |              |
| Azure OpenAI          | ➖             |              |
| xAI (Grok)           | ➖             |              |
```

## Stable Release Criteria

For embedding components run in the Spice runtime:
- Can handle X TPS (Transactions Per Second) and Y tokens per second successfully and without increasing latency for 1 hour.

### Documentation

- [ ] Documentation includes all steps to set up the embedding component.
- [ ] Documentation includes known limitations or issues for the embedding component.
- [ ] Documentation includes any exceptions made for Stable quality.

> - [ ] The embedding component has an easy-to-follow quickstart.

Reviewer: ```suggestion
- [ ] The embedding component has an easy-to-follow Cookbook Recipe.
```

# Spice.ai OSS Embeddings - Stable Release Criteria

This document defines the set of criteria that is required before an embedding component is considered to be of Stable quality.

All criteria must be met for the embedding component to be considered Stable.

## Stable Quality Embeddings

|     Embedding Type      | Stable Quality | DRI Sign-off |
| ----------------------- | -------------- | ------------ |
| File                    | ➖             |              |
| Hugging Face            | ➖             |              |
| Spice.ai Cloud Platform | ➖             |              |
| OpenAI                  | ➖             |              |
| Azure                   | ➖             |              |
| xAI (Grok)              | ➖             |              |

## Stable Release Criteria

> - All [RC release criteria](./rc.md) pass.

Reviewer 2: ```suggestion
- [ ] All [RC release criteria](./rc.md) pass.
```

### Conditional Criteria
- An embedding model running in the Spice runtime (as opposed to network-attached models), can handle consistent requests from several clients without an adverse impact on latency.

>   - N clients consistently sending requests (i.e. sending another request upon reciept of prior request)

Reviewer 3: ```suggestion
  - N clients consistently sending requests (i.e. sending another request upon receipt of prior request)
```

>   - N clients consistently sending requests (i.e. sending another request upon reciept of prior request)

Reviewer 3: Can we define `N`?

  - A duration of 60 minutes.

>   - The body must have at least Y tokens.

Reviewer 3: Can we define Y?

> - An embedding model running in the Spice runtime (as opposed to network-attached models), can handle consistent requests from several clients without an adverse impact on latency.
>   - N clients consistently sending requests (i.e. sending another request upon reciept of prior request)
>   - A duration of 60 minutes.
>   - The body must have at least Y tokens.
>   - An increase in latency is defined as a 10% increase in both the 50th & 95th percentile between the first and last minute.

Reviewer 2: ```suggestion
- [ ] An embedding model running in the Spice runtime (as opposed to network-attached models), can handle consistent requests from several clients without an adverse impact on latency.
  - [ ] N clients consistently sending requests (i.e. sending another request upon reciept of prior request)
  - [ ] A duration of 60 minutes.
  - [ ] The body must have at least Y tokens.
  - [ ] An increase in latency is defined as a 10% increase in both the 50th & 95th percentile between the first and last minute.
```
